filebeat/reader/message.go

```diff
@@ -49,13 +49,28 @@ func (m *Message) IsEmpty() bool {
 	return false
 }
 
-func (msg *Message) AddFields(fields common.MapStr) {
+// AddFields adds fields to the message.
+func (m *Message) AddFields(fields common.MapStr) {
 	if fields == nil {
 		return
 	}
 
-	if msg.Fields == nil {
-		msg.Fields = common.MapStr{}
+	if m.Fields == nil {
+		m.Fields = common.MapStr{}
 	}
-	msg.Fields.Update(fields)
+	m.Fields.Update(fields)
+}
+
+// AddTagsWithKey adds tags to the message with an arbitrary key.
+// If the field does not exist, it is created.
```

Comment:
We should name it in the code the same as it's in the event again. If we stay with `status` this should be `AddStatusWithKey`. Alternative as proposed above we could call it flags so it would be `AddFlagsWithKey`. Or if we stick with `tags` here it should probably be `log.tags`?

Would `AddFlagWithKey(key string, flag string) error` be enough here? It seems all the usage we have below is only adding 1 flag.